File: tests/fixtures/sample.rb
class GraphWalker
  attr_reader :nodes

  def initialize(nodes)
    @nodes = nodes
  end

  def buildMiniGraph(count)
    @nodes.each do |n|
      puts n
    end
  end

  def self.lazyBuildContext
    new([])
  end
end

walker = GraphWalker.new(['a', 'b'])
walker.buildMiniGraph(2)

File: tests/fixtures/sample.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    char* id;
    int value;
} GraphWalker;

GraphWalker* buildMiniGraph(char** nodes, int count) {
    GraphWalker* graph = malloc(count * sizeof(GraphWalker));
    for (int i = 0; i < count; i++) {
        graph[i].id = strdup(nodes[i]);
        graph[i].value = 1;
    }
    return graph;
}

void walk(GraphWalker* walker, char* startId, int depth) {
    (void)walker;
    (void)startId;
    (void)depth;
}

char** lazyBuildContext(char* query, int* count) {
    (void)query;
    (void)count;
    return NULL;
}

#define MAX_RESULTS 100
#define DEFAULT_DEPTH 2

int main() {
    char* nodes[] = {"a", "b"};
    int count = sizeof(nodes) / sizeof(nodes[0]);
    GraphWalker* g = buildMiniGraph(nodes, count);
    free(g);
    return 0;
}
